classes/zwitter.rb

```diff
@@ -13,31 +21,513 @@ def initialize
+        display_profile(zombie)
+        profile_menu(zombie)
+      # 6: Logout
+      else
+        puts "LOGGING OUT"
+        puts ""
+        user_men_sel = false
+#        show_main_menu()
+      end
+    end
+  end
+
+  # Displays and operates the interaction menu
+  def interact_menu(zombie)
+    interact_men_sel = true
+    while interact_men_sel
+      puts "INTERACTION MENU:"
+      puts ""
+      interact_men_sel = get_menu_selection(INTERACT_MENU)
+      case interact_men_sel
+      # 1: Create Tweet
+      when 1
+        content = get_input("What's on your mind (besides brains)?")
+        tweet = zombie.create_tweet(content: content)
+        display_tweet_feed(zombie)
+      # 2: Retweet
+      when 2
+        display_tweet_feed(zombie)
+        print "TWEET ID TO RETWEET: "
+        tweet_id = gets.chomp.to_i
+        puts ""
+        tweet_to_retweet = Tweet.find_tweet(tweet_id)
+        if tweet_to_retweet
+          retweet = zombie.retweet(tweet_id)
+          puts "RETWEETED:"
+          puts ""
+          show_single_tweet(zombie, retweet)
+        else
+          puts "NO TWEET EXISTS WITH THIS ID"
+          puts ""
+        end
+      # 3: Delete Tweet
+      when 3
+        display_tweets(zombie)
+        print "TWEET ID TO DELETE: "
+        tweet_id = gets.chomp.to_i
+        puts ""
+        tweet_to_delete = zombie.return_tweet(tweet_id)
+        if tweet_to_delete
+            puts "YOU SELECTED THIS TWEET:"
+            puts ""
+            show_single_tweet(zombie, tweet_to_delete)
+          confirm = get_input("CONFIRM DELETE, TYPE Y OR N").downcase
+          if confirm == "y"
+            zombie.delete_tweet(tweet_id)
+            puts "TWEET DELETED"
+            puts ""
+            display_tweet_feed(zombie)
+          else
+            puts "DELETE CANCELLED"
+            puts ""
+            display_tweet_feed(zombie)
+          end
+        else
+          puts "YOU DO NOT HAVE A TWEET WITH THIS ID"
+          puts ""
+        end
+      # 4: Fave Tweet
+      when 4
+        display_tweet_feed(zombie)
+        print "TWEET ID TO FAV: "
+        tweet_id = gets.chomp.to_i
+        puts ""
+        tweet = Tweet.find_tweet(tweet_id)
+        if tweet
+          already_favd = false
+          tweet.favs.each do | fav |
+            if fav.zombie.username == zombie.username
+              already_favd = true
+            end
+          end
+          if !already_favd
+            tweet.add_to_favs(zombie)
+            display_tweet_feed(zombie)
+          else
+            puts "YOU ALREADY FAV'D THIS TWEET."
+            puts ""
+          end
+        else
+          puts "TWEET ID DOES NOT EXIST"
+          puts ""
+          display_tweet_feed(zombie)
+        end
+      # 5: Unfave Tweet
+      when 5
+        display_tweet_feed(zombie)
+        print "TWEET ID TO UNFAV: "
+        tweet_id = gets.chomp.to_i
+        puts ""
+        tweet = Tweet.find_tweet(tweet_id)
+        if tweet
```

Comment:
since you're not reusing this variable, you could just do
```ruby
if Tweet.find_tweet(tweet_id)
```
though this is discretionary